[Zheng Kaiwen] ip

Kevin88866 opened the pull request:
complete level-0,1,2,3,4

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Kevin88866 committed:
Level 0. Rename, Greet, Exit

Kevin88866 committed:
Merge branch 'add-gradle-support'

Kevin88866 committed:
feat: implement echo function and refactor message printing

- implement echo loop that repeats user input until "bye"
- Extracted message printing into sendMessage() for better code reuse

Kevin88866 committed:
Change function name to MessagePrinter and add sc.close()

Kevin88866 committed:
feat: add add,list funtionality

Kevin88866 committed:
feat: add mark and unmark as done functionality and create Task class

- add Task class, include task and isDone
- use ArrayList to store tasks

Kevin88866 committed:
Improve code quality by refactoring

Kevin88866 committed:
Complete ToDOs, Event, Deadlines

Create abstract class tasks, and override the toString() in class todo,deadline and event. In this case, all type of tasks can be print by printTask(task).

Kevin88866 committed:
Add Text UI Testing

Pavithra6-Srinivasan commented on src/main/java/kiki.java:
```diff
@@ -0,0 +1,87 @@
+        Scanner sc = new Scanner(System.in);
+        MessagePrinter(" Hello! I'm kiki\n  What can I do for you?");
+        String input;
+        while(!(input = sc.nextLine()).equals("bye")){
```
I like that the if statements are clear and organised well

gohjieling834 commented on src/main/java/kiki.java:
```diff
@@ -0,0 +1,87 @@
+        MessagePrinter(" Hello! I'm kiki\n  What can I do for you?");
+        String input;
+        while(!(input = sc.nextLine()).equals("bye")){
+            if(input.equals("list")){
+                printList();
+            }else if(input.startsWith("mark")){
+                markTask(input);
+            }else if(input.startsWith("unmark")){
+                unmarkTask(input);
+            }else if(input.startsWith("todo")){
+                addTodo(input);
+            }else if(input.startsWith("deadline")){
+                addDeadline(input);
+            }else if(input.startsWith("event")){
+                addEvent(input);
+            }
```
I like that you created different methods for the different functionalities. Looks very neat and easy to follow.

gohjieling834 replied:
Overall, your code is neat and easy to understand. Keep up the good work!

Pavithra6-Srinivasan commented on src/main/java/kiki.java:
```diff
@@ -0,0 +1,87 @@
+    private static void printList() {
+        System.out.println(horizontalLine);
+        System.out.println("  Here are the tasks in your list:");
+        for(int i = 0; i < tasks.size(); i++) System.out.println("  " + (i+1) + ". " + tasks.get(i).toString());
```
Do you think this should be in a single line?

Pavithra6-Srinivasan replied:
You can add a space between for and bracket

Pavithra6-Srinivasan commented on src/main/java/kiki.java:
```diff
@@ -0,0 +1,87 @@
+        MessagePrinter(" Hello! I'm kiki\n  What can I do for you?");
+        String input;
+        while(!(input = sc.nextLine()).equals("bye")){
+            if(input.equals("list")){
+                printList();
+            }else if(input.startsWith("mark")){
+                markTask(input);
+            }else if(input.startsWith("unmark")){
+                unmarkTask(input);
+            }else if(input.startsWith("todo")){
+                addTodo(input);
+            }else if(input.startsWith("deadline")){
+                addDeadline(input);
+            }else if(input.startsWith("event")){
```
You can add a space between the bracket and else if statement for more clarity

Pavithra6-Srinivasan reviewed (commented):
Overall, code is understandable.

gohjieling834 commented on src/main/java/kiki.java:
```diff
@@ -0,0 +1,87 @@
+        sc.close();
+    }
+
+    private static void MessagePrinter(String message){
```
I noticed some of the methods or while loop is missing a space between the ')' and '{'. Ctrl + Alt + L can help you format it. :)

gohjieling834 replied:
Maybe you can consider printMessage as the method name, so that the method naming is standardised.

gohjieling834 commented on src/main/java/kiki.java:
```diff
@@ -0,0 +1,87 @@
+import java.util.ArrayList;
+
+public class kiki {
+    private static final String horizontalLine = "____________________________________________________________";
```
Constant names should be using SCREAMING_SNAKE_CASE. Perhaps you can consider using DIVIDER as the variable name?

Kevin88866 committed:
centralize command handling and add exception

- introduce KikiException for user-facing errors
- validate todo/deadline/event formats and mark/unmark indices
- add unified parseIndex with clear messages
- wrap main loop with try/catch for friendly error printing

Kevin88866 committed:
Modify input.txt and EXPECTED.TXT

Kevin88866 committed:
Merge branch 'branch-Level-5'

Kevin88866 committed:
Organize the classes into suitable java packages.

Kevin88866 committed:
Merge branch 'branch-A-Packages'

Kevin88866 committed:
add some comments to make it easier to understand what each class and function does

roshnidaksha commented on src/main/java/kiki/task/Deadline.java:
```diff
@@ -0,0 +1,17 @@
+package kiki.task;
+// class Deadline
+// format: deadline ___ /by ___
```
Format your comments to follow CS2113 coding standards.

roshnidaksha commented on src/main/java/kiki/task/Deadline.java:
```diff
@@ -0,0 +1,17 @@
+    }
+
+    @Override
+    public String toString(){
```
Leave spaces before `{`. The same applies to your other methods as well.

roshnidaksha commented on src/main/java/kiki/task/Event.java:
```diff
@@ -0,0 +1,18 @@
+//format: event ___ /from ___ /to ___
+
+public class Event extends Task{
+    private String from,to;
```
Add spaces between punctuators and operators to make it more readable.

roshnidaksha commented on src/main/java/kiki/ui/kiki.java:
```diff
@@ -0,0 +1,208 @@
+        MessagePrinter(" Hello! I'm Kiki.kiki\n  What can I do for you?");
+        String input;
+        //the loop will never stop until the input is "bye"
+        while(!(input = sc.nextLine()).equals("bye")){
```
Avoid giving comments to code that is self-explanatory.

roshnidaksha commented on src/main/java/kiki/ui/kiki.java:
```diff
@@ -0,0 +1,208 @@
+            }catch(KikiException e){
+                //print the message in KikiException
+                MessagePrinter(e.getMessage());
+            }catch(Exception e){
```
Catching `Exception` is quite generic, consider replacing it with some specific examples.

roshnidaksha commented on src/main/java/kiki/ui/kiki.java:
```diff
@@ -0,0 +1,208 @@
+
+    //handle input
+    //will throw KikiException when something get wrong
+    private static void handle(String input) throws KikiException{
```
Perhaps `handleCommand` would be a better name here.

roshnidaksha commented on src/main/java/kiki/ui/kiki.java:
```diff
@@ -0,0 +1,208 @@
+        tasks.add(task);
+        printTask(task);
+    }
+}
```
Try extracting out related data/methods into a separate class to make this shorter.

roshnidaksha reviewed (commented):
Overall, well written.
There are spaces missing between parenthesis at many places and some minor suggestions. Do take a look at my comments below.

Kevin88866 committed:
Add support for deleting tasks from the list.

Kevin88866 committed:
change the input.txt, epected.txt and runtest.bat

- add delete function to the input.txt
- change runtest.bat since I have packaged the code, so the location of my code change

Kevin88866 committed:
Save and load the tasks in the hard disk.

Kevin88866 committed:
Merge branch 'branch-Level-6'

Kevin88866 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/kiki/ui/kiki.java

Kevin88866 committed:
add saveList(storage data) in the delete function

Kevin88866 committed:
Package the app as an executable JAR file

Kevin88866 committed:
fix code style issues to pass checkstyle

Kevin88866 committed:
Refactor the code to extract out closely related code as classes

Kevin88866 committed:
add LocalDate support and "on" command for date queries

- Store deadline/event dates as java.time.LocalDate
- Format output using MMM d yyyy (e.g., Oct 15 2019)
- Keep save/load in ISO yyyy-MM-dd format for backward compatibility
- Added OnDateCommand to list tasks occurring on a specific date
- Updated Deadline and Event with toSaveString() using ISO date format

Kevin88866 committed:
Add find function so that users can find a task by searching for a keyword in the task description

Kevin88866 committed:
Add a User Guide to the project

Kevin88866 committed:
Package the app as an executable JAR file and update the kikiTest

Kevin88866 committed:
update readme

Kevin88866 committed:
update readme

Kevin88866 committed:
update readme

Kevin88866 committed:
Merge pull request #1 from Kevin88866/branch-Readme

update readme

Kevin88866 committed:
update readme

Kevin88866 committed:
Merge branch 'branch-Readme'

Kevin88866 committed:
Merge branch 'master' of github.com:Kevin88866/ip

Kevin88866 committed:
Update README.md

Kevin88866 committed:
Update README.md

Kevin88866 committed:
Update README.md
